### PR TITLE
Loopy sprint

### PR DESCRIPTION
**Pull request by ReubenHill**
Linked to https://github.com/firedrakeproject/firedrake/pull/2038 and https://github.com/OP2/PyOP2/pull/612. Requires loopy branch https://github.com/firedrakeproject/loopy/tree/kernel_callables_v3-edit2

### CODE REVIEW

**Comment by sv2518** [comment]
This is ready to be reviewed. @ReubenHill  could you please convert into an actual (not draft) PR?

**Comment by ReubenHill** [comment]
> This is ready to be reviewed. @ReubenHill could you please convert into an actual (not draft) PR?

done

File: tsfc/kernel_interface/firedrake_loopy.py [hunk 1]
```diff
@@ -42,6 +43,7 @@ class Kernel(object):
     :kwarg quadrature_rule: The finat quadrature rule used to generate this kernel
     :kwarg tabulations: The runtime tabulations this kernel requires
     :kwarg needs_cell_sizes: Does the kernel require cell sizes.
+    :kwarg name: What is the name of this kernel.
```
Comment (dham) [inline tsfc/kernel_interface/firedrake_loopy.py:46]: Lose "What is"

File: requirements-git.txt [hunk 2]
```diff
@@ -2,4 +2,4 @@ git+https://github.com/coneoproject/COFFEE.git#egg=coffee
 git+https://github.com/firedrakeproject/ufl.git#egg=fenics-ufl
 git+https://github.com/firedrakeproject/fiat.git#egg=fenics-fiat
 git+https://github.com/FInAT/FInAT.git#egg=finat
-git+https://github.com/firedrakeproject/loopy.git@firedrake#egg=loo.py
+git+https://github.com/firedrakeproject/loopy.git@kernel_callables_v3-edit2#egg=loopy
```
Comment (dham) [inline requirements-git.txt:5]: Drop before merging and change loopy at the same time.

**Review by dham — COMMENTED** [review]
Basically there.

**Pull request merged**